src/parser/SparqlParser.cpp

```diff
@@ -19,101 +19,18 @@ SparqlParser::SparqlParser(const string& query) : lexer_(query), query_(query) {
   LOG(DEBUG) << "Parsing " << query << std::endl;
 }
 
-namespace {
-// Converts the PrefixMap to the legacy data format used by ParsedQuery
-vector<SparqlPrefix> convertPrefixMap(
-    const SparqlQleverVisitor::PrefixMap& map) {
-  vector<SparqlPrefix> prefixes;
-  for (auto const& [label, iri] : map) {
-    prefixes.emplace_back(label, iri);
-  }
-  return prefixes;
-}
-}  // namespace
-
 // _____________________________________________________________________________
 ParsedQuery SparqlParser::parse() {
-  ParsedQuery result;
-  std::string originalString = query_;
-  // parsePrologue parses all the prefixes which are stored in a member
-  // PrefixMap. This member is returned on parse.
-  SparqlQleverVisitor::PrefixMap prefixes = parseWithAntlr(
-      &AntlrParser::prologue,
-      {{INTERNAL_PREDICATE_PREFIX_NAME, INTERNAL_PREDICATE_PREFIX_IRI}});
+  ParsedQuery result = parseWithAntlr(
+      &AntlrParser::query,
+      {{INTERNAL_PREDICATE_PREFIX_NAME, INTERNAL_PREDICATE_PREFIX_IRI}},
+      query_);
```

Comment:
I think in the meantime this should be a static function `SparqlParser::parseQuery(std::string_view query)` which doesn't use the `lexer` at all. 
I've seen that you have already change `parseWithAntlr` somewhat into this direction.